src/clang_parser.cpp

```diff
@@ -592,8 +592,11 @@ bool ClangParser::parse(ast::Program *program, BPFtrace &bpftrace, std::vector<s
     args.push_back(flag.c_str());
   }
 
-  bool process_btf = program->c_definitions.empty() ||
-                     (bpftrace.force_btf_ && bpftrace.btf_.has_data());
+  // Don't parse BTF for programs containing userspace probes if the user
+  // defined some custom data types, since those may conflict with BTF types.
+  bool process_btf = bpftrace.btf_.has_data() &&
+                     !(program->has_userspace_probes() &&
+                       !program->c_definitions.empty());
```

Comment:
Easier to read if we avoid double negative
```suggestion
  bool process_btf = bpftrace.btf_.has_data() &&
                     (!program->has_userspace_probes() ||
                       program->c_definitions.empty());
```